custom/src/main/java/org/kiwix/kiwixmobile/custom/main/CustomMainActivity.kt

```diff
@@ -79,8 +76,9 @@ class CustomMainActivity : CoreMainActivity() {
       .isVisible = false
   }
 
-  override fun openSettingsActivity() {
-    startActivityForResult(intent<CustomSettingsActivity>(), REQUEST_PREFERENCES)
+  override fun openSettings() {
+    super.openSettings()
+    navigate(R.id.customSettingsFragment)
```

Comment:
why are we not doing this how the others are done? with abstract vals and the implementation of the navigation in Core?

Reply:
Good question!